package.json

```diff
@@ -34,7 +34,9 @@
       "<rootDir>test/setupTests.js"
     ],
     "moduleNameMapper": {
-      "\\.(css|less|scss)$": "identity-obj-proxy"
+      "\\.(css|less|scss)$": "identity-obj-proxy",
+      "\\.svg$": "<rootDir>/__mocks__/svgMock.js",
+      "\\.(jpg|jpeg|png|gif|eot|otf|webp|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js"
```

Comment:
It seems that some of our projects (e.g. CLUE) use `__mocks__` consistently for this purpose, while others use `identity-object-proxy` consistently, and other use a mix of both (like this). Do you have a rule-of-thumb for when to prefer one over the other?

Reply:
My understanding is that they're not really interchangeable. `identity-object-proxy` works nicely with CSS modules, as all the named imports are changed to strings and we can even use that in tests. For example:
```
import css from "./styles"

<div className={css.header}>
```
becomes
```
<div className="header">
```




It be nice if it worked for other assets like images, but I've tried and it doesn't:
<img width="451" alt="Screenshot 2021-07-05 at 10 34 48" src="https://user-images.githubusercontent.com/767857/124443188-c2930e80-dd7d-11eb-94fa-65519e7f1709.png">

I guess it's because assets use default imports. That's why I'm using custom files/svg mocks. But if there's a way to use `identity-object-proxy` for everything, I'd happy to do it.

Reply:
@eireland ran into this recently in [portal-report #344](https://github.com/concord-consortium/portal-report/pull/344) where we added the
```
"\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "identity-obj-proxy"
```
to fix an error with PNG import, but we only did it that way because that's the way the styles were being imported. In that limited case the `identity-object-proxy` seemed to work, but it sounds like that's not always the case. Given your experience it sounds like `identity-object-proxy` is preferred for styles because it works better with CSS Modules (although this is less important in projects that aren't using CSS Modules, like CLUE), while `__mocks__` is preferred for assets because it works in some situations where `identity-object-proxy` does not (although those limitations apparently don't affect all projects, e.g. portal-report). Thanks for the explanation.